geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/data/KafkaTruncateTopic.scala

```diff
@@ -0,0 +1,216 @@
+      val configResource = new ConfigResource(ConfigResource.Type.TOPIC, topicName)
+      val configsResult = admin.describeConfigs(Collections.singleton(configResource))
+      val config = configsResult.all().get().get(configResource)
+
+      config.get(TopicConfig.CLEANUP_POLICY_CONFIG).value().contains(CLEANUP_POLICY_DELETE)
+    } catch {
+      case _: InterruptedException =>
+        Thread.currentThread().interrupt()
+        throw new RuntimeException("describeTopics operation interrupted, deleteRecords will not executed.")
+      case e: ExecutionException =>
+        throw convertExecutionException(e)
+    }
+
+  /**
+   * Add the 'delete' cleanup policy to the topic's 'cleanup.policy' config.
+   */
+  private def addDeleteCleanupPolicy(topic: String): Unit =
+    try {
+      val configResource = new ConfigResource(ConfigResource.Type.TOPIC, topic)
+      val alterConfigOp = new AlterConfigOp(new ConfigEntry(CLEANUP_POLICY, CLEANUP_POLICY_DELETE), AlterConfigOp.OpType.APPEND)
+      val configs: util.Map[ConfigResource, util.Collection[AlterConfigOp]] = Map(configResource -> alterConfigOpColl(alterConfigOp)).asJava
+      admin.incrementalAlterConfigs(configs).all().get()
+    } catch {
+      case _: InterruptedException =>
+        Thread.currentThread().interrupt()
+        throw new RuntimeException("incrementalAlterConfigs operation interrupted, ...")
+      case e: ExecutionException =>
+        throw convertExecutionException(e)
+    }
+
+  /**
+   * Remove the 'delete' cleanup policy to the topic's 'cleanup.policy' config.
+   */
+  private def removeDeleteCleanupPolicy(topic: String): Unit =
+    try {
+      val configResource = new ConfigResource(ConfigResource.Type.TOPIC, topic)
+      val alterConfigOp = new AlterConfigOp(new ConfigEntry(CLEANUP_POLICY, CLEANUP_POLICY_DELETE), AlterConfigOp.OpType.SUBTRACT)
+      val configs: util.Map[ConfigResource, util.Collection[AlterConfigOp]] = Map(configResource -> alterConfigOpColl(alterConfigOp)).asJava
+      admin.incrementalAlterConfigs(configs).all().get()
+    } catch {
+      case _: InterruptedException =>
+        Thread.currentThread().interrupt()
+        throw new RuntimeException("incrementalAlterConfigs operation interrupted, was not able to remove the 'delete' cleanup.policy.")
+      case e: ExecutionException =>
+        throw convertExecutionException(e)
+    }
+
+  /**
+   * Singleton wrapper for AlertConfigOp.
+   */
+  private def alterConfigOpColl(alterConfigOp: AlterConfigOp): util.Collection[AlterConfigOp] =
+    Collections.singleton(alterConfigOp)
+
+  /**
+   * Convert ExecutionException to RuntimeException, preserving the underlying cause.
+   */
+  private def convertExecutionException(e: ExecutionException): RuntimeException = {
+    val cause = Option(e.getCause).getOrElse(e)
+    new RuntimeException(cause.getMessage, cause)
+  }
+}
+
+object KafkaTruncateTopic {
+  val CLEANUP_POLICY = "cleanup.policy"
+  val CLEANUP_POLICY_DELETE = "delete"
```

Comment:
can use `org.apache.kafka.common.config.TopicConfig#CLEANUP_POLICY_CONFIG` and `org.apache.kafka.common.config.TopicConfig#CLEANUP_POLICY_DELETE`